### PR TITLE
Button padding should use size "medium" by default

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR ensures that StyledButtonKind uses size "medium" by default. Currently, it wasn't falling back to any size and this was causing discrepancies between buttons implemented as `<Button />` vs `<Button size="medium" />` but they should be identical.

#### Where should the reviewer start?
src/js/components/Button/StyledButtonKind.js

#### What testing has been done on this PR?
Tested local in storybook, created custom theme with size definitions. Default (no size prop specified) and medium should be the same always.

Before (notice padding difference between default and medium):
Screen Shot 2021-02-17 at 3 52 14 PM

After:
Screen Shot 2021-02-17 at 3 52 23 PM

#### Any background context you want to provide?
Noticed the HPE button dimensions had shifted after we removed padding values in this PR but when I looked into it further I also noticed that buttons implemented as `<Button label="Label" />` were not receiving styling for "medium" size like they should.

#### What are the relevant issues?
Noticed issue after https://github.com/grommet/grommet-theme-hpe/pull/163 merged in HPE theme and a subsequent PR on design system site noted snapshot changes.

#### Screenshots (if appropriate)
See above.

#### Do the grommet docs need to be updated?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/Button/__tests__/Button-kind-test.js [hunk 1]
```diff
@@ -381,4 +381,53 @@ describe('Button kind', () => {
     fireEvent.mouseOver(getByText('Button'));
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test(`should apply size styling correctly`, () => {
```
Comment (halocline) [inline src/js/components/Button/__tests__/Button-kind-test.js:385]: What is "correctly"?

```suggestion
  test(`should apply styling according to theme size definition`, () => {
```

**Review by halocline — COMMENTED** [review]
One wording suggestion for test name, the rest looks good.

**Review by halocline — APPROVED** [review]

**Comment by halocline** [comment]
@taysea looks like the test name update needs the associated snapshot updated.

File: src/js/components/Button/StyledButtonKind.js [hunk 2]
```diff
@@ -32,7 +32,8 @@ const fontStyle = props => {
   `;
 };
 
-const padFromTheme = (size, theme) => {
+const padFromTheme = (sizeProp, theme) => {
```
Comment (ericsoderberghp) [inline src/js/components/Button/StyledButtonKind.js:35]: FYI, I think you could have just done: `(size = 'medium', theme)`

**Review by MikeKingdom — APPROVED** [review]
Looks good and I agree with Eric, you can probably use the arg default mechanism.

Sorry I missed that. I thought I remembered it doing that up higher in Button itself.

File: src/js/components/Button/StyledButtonKind.js (hunk 2 above)
Comment (taysea) [reply src/js/components/Button/StyledButtonKind.js:35]: Good reminder, updating now.

**Comment by taysea** [comment]
@ericsoderberghp @MikeKingdom @halocline I've adjusted the PR to define "medium" as the fallback size directly in Button.js. I noticed that in `StyledButton` for themes without any `theme.button.default` the same discrepancies were also showing (see screenshots). I think setting size higher up is a smoother solution:

For this I defined some padding/border radius for the various sizes, but notice how it's not carried through to the Default Button (`<Button />`)

Screen Shot 2021-02-18 at 8 38 18 AM

**Review by MikeKingdom — COMMENTED** [review]
Coolness. Makes sense to me

**Comment by taysea** [comment]
> Coolness. Makes sense to me

Hmm need to look at some chromatic changes actually.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**